c/src/sbp.c

```diff
@@ -566,23 +536,25 @@ s8 sbp_process(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void *context))
     break;
 
   case SBP_GET_CRC:
-    ret = sbp_state_read_to_frame_buffer(s, read, SBP_CRC_LEN - s->n_read);
+    ret = sbp_state_read_to_frame_buffer(s, read, (u8)(SBP_CRC_LEN - s->n_read));
     if (ret != SBP_OK) {
       return ret;
     }
     if (s->n_read >= SBP_CRC_LEN) {
       s->state = SBP_WAITING;
       s->crc = sbp_u8_array_to_u16(&(s->frame_buff[SBP_FRAME_OFFSET_CRC(s->msg_len)]));
 
-      sbp_u16_to_u8_array(s->msg_type, (u8*)&buf);
+      u16 raw_msg_type = (u16)s->msg_type;
+      sbp_u16_to_u8_array(raw_msg_type, (u8*)&buf);
+      s->msg_type = (sbp_msg_type_t)raw_msg_type;
```

Comment:
`s->msg_type` has changed to be type `sbp_msg_type_t` rather than `u16`, and the size of an enum is implementation defined. It needs to be copied to a temp variable which is definitely `u16` before trying to encode it.

Reply:
Sorry, was actually asking about line 549 - do we need to copy it back?

Reply:
Oh I see, yeah I think you're right that doesn't need to be there